src/saveload/saveload.cpp

```diff
@@ -1620,6 +1639,135 @@ void SlObject(void *object, const SaveLoadTable &slt)
 	}
 }
 
+/**
+ * Save or Load a table header.
+ * Note: a table-header can never contain more than 65535 fields.
+ * @param slt The SaveLoad table with objects to save/load.
+ * @return When loading, the ordered SaveLoad array to use; otherwise an empty list.
+ */
+std::vector<SaveLoad> SlTableHeader(const SaveLoadTable &slt)
+{
+	switch (_sl.action) {
+		case SLA_LOAD_CHECK:
+		case SLA_LOAD: {
+			std::vector<SaveLoad> saveloads;
+
+			/* Build a key lookup mapping based on the available fields. */
+			std::map<std::string, const SaveLoad *> key_lookup;
+			for (auto &sld : slt) {
+				key_lookup[sld.name] = &sld;
+			}
+
+			while (true) {
+				static std::string key;
```

Comment:
Does this need to be static? The SLEG_SSTR is not.

Reply:
`saveload/saveload.cpp:1683:47: error: ‘key’ is not captured`.

I really did not bother looking into what is going on exactly, as this rabbit hole is deep enough as it is :D From what I understand, how `SLEG_` is designed, it needs global.

Reply:
Thanks, makes sense.